src/pybind/mgr/dashboard/frontend/src/app/shared/api/rgw-bucket.service.ts

```diff
@@ -65,25 +65,28 @@ export class RgwBucketService extends ApiClient {
     cannedAcl: string
   ) {
     return this.rgwDaemonService.request((params: HttpParams) => {
+      const paramsObject = {
+        bucket,
+        uid,
+        zonegroup,
+        lock_enabled: String(lockEnabled),
+        lock_mode,
+        lock_retention_period_days,
+        encryption_state: String(encryption_state),
+        encryption_type,
+        key_id,
+        tags: tags,
+        bucket_policy: bucketPolicy,
+        canned_acl: cannedAcl,
+        daemon_name: params.get('daemon_name')
+      };
+
+      if (placementTarget) {
+        paramsObject['placement_target'] = placementTarget;
+      }
```

Comment:
it checks for `placementTarget` L84 and then sends request.
So its not setting `Null` in any case

Reply:
Ack. Not even empty quotes or list?

Reply:
for empty string api accepts that, not for list.
`null` is used at other places in code to set default values, also for checks so just kept that as it is.

https://github.com/ceph/ceph/blob/cab8141bf179785b760a9c73b3990a597f8a8a8e/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts#L119-L136

https://github.com/ceph/ceph/blob/cab8141bf179785b760a9c73b3990a597f8a8a8e/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.html#L109-L117

Reply:
So if null is used by default, then do we need the if condition?

Reply:
We need if condition because `null` values are not accepted by api. if you will chekc it will give you Invalud location constraint.
Thats something the API design and the methodology that went to design it like that to allow/disallow nulls in JSON

Additionally, placement_target is an optional value in api, so omitting it would be fine here.

Reply:
sorted offline in slack! hence resolving!